README.md

```diff
@@ -1 +1,303 @@
+
+![](/images/em-intro.png)
+
+
+
+## Advantage
+
+EdgeMesh satisfies the new requirements in edge scenarios (e.g., limited edge resources, unstable edge cloud network, etc.), that is, high availability, high reliability, and extreme lightweight:
+
+- **High availability**
+  - Open up the network between edge nodes by using the edge cloud channel in KubeEdge
+  - Divide the communication between edge nodes into intra-LAN and cross-LAN
+    - Intra-LAN communication: direct access
+    - Cross-LAN communication: forwarding through the cloud
+- **High reliability (offline scenario)**
+  - Both control plane and data plane traffic are delivered through the edge cloud channel
+  - EdgeMesh internally implements a lightweight DNS server, thus no longer accessing the cloud DNS
+- **Extreme lightweight**
+  - Each node has one and only one EdgeMesh, which saves edge resources
+
+#### User value
+
+- For edge devices with limited resources, EdgeMesh provides a lightweight and highly integrated software with service discovery
+- In the scene of Field Edge, compared to the mechanism of coredns + kube-proxy + cni service discovery , users only need to simply deploy an EdgeMesh to finish their goals
+
+
+
+## Architecture
+
+![](/images/em-arch.png)
+
+To ensure the capability of service discovery in some edge devices with low-version kernels or low-version iptables, EdgeMesh adopts the userspace mode in its implementation. In addition, it also comes with a lightweight DNS server.
+
+- Through the capability of list-watch on the edge of KubeEdge, EdgeMesh monitors the addition, deletion and modification of metadata (e.g., services and endpoints), and then creates iptables rules based on services and endpoints
+- EdgeMesh uses the same ways (e.g., Cluster IP, domain name) as the K8s Service to access services
+- When client's requests accessing a service reach a node with EdgeMesh, it will enter the kernel's iptables at first
+- The iptables rules previously configured by EdgeMesh will redirect requests, and forward them all to the port 40001 which is occupied by the EdgeMesh process (data packets from kernel mode to user mode)
+- After requests enter the EdgeMesh process, the EdgeMesh process completes the selection of backend Pods (load balancing occurs here), and then sends requests to the host where the Pod is located
+
+
+
+
+## Functionality
```

Comment:
Key features and roadmap

Reply:
I have modified this content as required.